src/main/java/dev/jbang/source/builders/BaseBuilder.java

```diff
@@ -131,9 +131,9 @@ public IntegrationResult compile() throws IOException {
 		optionList.addAll(prj.getMainSourceSet().getCompileOptions());
 		String path = prj.resolveClassPath().getClassPath();
 		if (!Util.isBlankString(path)) {
-			optionList.addAll(Arrays.asList("-classpath", path));
+			optionList.addAll(Arrays.asList("-classpath", escapeOSArgument(path, shell)));
```

Comment:
I left a bunch of comments on #1447 about some tests I did and I think this fix only works for Windows but breaks things on Linux/Mac.